src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -442,34 +510,286 @@ public int getID()
+        if (Carrots >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvestOneCarrot);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest25Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest100Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest1000Carrots);
+        }
+    }
+    /**
+     * Increment the potatoes harvested by this colony.
+     * Will award achievements for potatoes harvested.
+     */
+    public void incrementPotatoesHarvested()
+    {
+        harvestedPotatoes++;
+        final int Potatoes = this.getHarvestedPotatoes();
+        if (Potatoes >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvestOnePotato);
+        }
+        if (Potatoes >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest25Potatoes);
+        }
+        if (Potatoes >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest100Potatoes);
+        }
+        if (Potatoes >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+        }
+        if (Potatoes >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest1000Potatoes);
+        }
+    }
+    /**
+     * Increment the wheat harvested by this colony.
+     * Will award achievements for wheat harvested.
+     */
+    public void incrementWheatHarvested()
+    {
+        harvestedWheat++;
+        final int Wheat = this.getHarvestedWheat();
+        if (Wheat >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvestOneWheat);
+        }
+        if (Wheat >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest25Wheat);
+        }
+        if (Wheat >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest100Wheat);
+        }
+        if (Wheat >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+        }
+        if (Wheat >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest1000Wheat);
+        }
+    }
+    public void incrementTreesFelled()
+    {
+        felledTrees++;
+        final int Trees = this.getFelledTrees();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)